base/abstractarray.jl

```diff
@@ -1737,6 +1737,7 @@ function mapslices(f, A::AbstractArray, dims::AbstractVector)
 
     Aslice = A[idx...]
     r1 = f(Aslice)
+    safe_for_reuse = isa(r1, Number) || (isa(r1, AbstractArray) && eltype(r1) <: Number)
```

Comment:
immutable and doesn't contain anything (and happens to cover the vast majority of what people use `mapslices` in practice for).